clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1322,12 +1322,11 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      */
     @Override
     public void seek(TopicPartition partition, long offset) {
-        acquireAndEnsureOpen();
         try {
             if (offset < 0)
                 throw new IllegalArgumentException("seek offset must not be a negative number");
-
             log.debug("Seeking to offset {} for partition {}", offset, partition);
+            acquireAndEnsureOpen();
```

Comment:
This needs to go outside of the `try` because the call to `release()` is in the `finally`. If we're going to do this, I would suggest moving the validation checks outside of the `try` as well.